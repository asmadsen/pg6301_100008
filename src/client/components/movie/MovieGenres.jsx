import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core'
import Chip from '@material-ui/core/Chip'

const useStyles = makeStyles({
	chip: {
		marginRight: 5
	}
})

const MovieGenres = ({ genres }) => {
	const classes = useStyles()

	return <div>
		{
			genres.map(genre => <Chip
				key={genre}
				label={genre}
				size="small"
				className={classes.chip}
			/>)
		}
	</div>
}

MovieGenres.propTypes = {
	genres: PropTypes.arrayOf(PropTypes.string).isRequired
}

export default MovieGenres
