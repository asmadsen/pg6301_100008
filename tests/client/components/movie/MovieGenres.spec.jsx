import React from 'react'
import { shallow } from 'enzyme'
import MovieGenres from '../../../../src/client/components/movie/MovieGenres'

describe('client/component/movie/MovieCard', () => {
	it('should match snapshot', () => {
		const genres = ['Action', 'Drama']
		const wrapper = shallow(<MovieGenres genres={genres} />)

		expect(wrapper).toMatchSnapshot()
	})
})
