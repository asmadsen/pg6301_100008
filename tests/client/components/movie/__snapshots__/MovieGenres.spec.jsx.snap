// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client/component/movie/MovieCard should match snapshot 1`] = `
<div>
  <WithStyles(ForwardRef(Chip))
    className="makeStyles-chip-1"
    key="Action"
    label="Action"
    size="small"
  />
  <WithStyles(ForwardRef(Chip))
    className="makeStyles-chip-1"
    key="Drama"
    label="Drama"
    size="small"
  />
</div>
`;
